### ISSUE TITLE
AWS SecretsManager JSON number value results in "unable to unmarshal secret"

### ISSUE THREAD

**Issue by jtut1731** [external-secrets/external-secrets#510]
**Describe the solution you'd like**
If a secret contains a JSON number value, this value should be converted to a string value.

**What is the added value?**
AWS SecretsManager includes the ability to automatically rotate RDS credentials. The credentials secret that it creates include "port", and the value is a number in JSON.

AWS SecretManager Secret (JSON):
```
{
  "database": "my-db",
  "password": "%K>U~(_b.&RJs#xFE!P-B?tmWDp3JXcd",
  "masterarn": "arn:aws:secretsmanager:us-east-1:1234567890:secret:secretMySqlMasterUser-ABC-ABC",
  "engine": "mysql",
  "port": 3306,
  "dbInstanceIdentifier": "abc123",
  "host": "abc123.us-east-1.rds.amazonaws.com",
  "username": "my-db-user"
}
```

should produce Secret data:
```
  database: "my-db",
  password: "%K>U~(_b.&RJs#xFE!P-B?tmWDp3JXcd",
  masterarn: "arn:aws:secretsmanager:us-east-1:1234567890:secret:secretMySqlMasterUser-ABC-ABC",
  engine: "mysql",
  port: "3306",
  dbInstanceIdentifier: "abc123",
  host: "abc123.us-east-1.rds.amazonaws.com",
  username: "my-db-user"
```

**Observations (Constraints, Context, etc):**
In code: /pkg/provider/aws/secretsmanager/secretsmanager.go: 117
Error message: secret my-secret: json: cannot unmarshal number into Go value of type string

**Comment by moolen** [comment]
Right, this is a bug! The `map[string]string` [here](https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/aws/secretsmanager/secretsmanager.go#L114) really should be a `map[string]interface{}` and then we have to do a type assertion and return the value as `[]byte`

**Comment by knelasevero** [comment]
I think we have this in some other providers as well

**Comment by nikonet** [comment]
> I think we have this in some other providers as well

It also happens with the GCP(SM) provider.

Related to this - parsing JSON nested objects (and arrays?) results in `unable to unmarshal secret: json: cannot unmarshal object into Go value of type string`. 

If it is possible to get around this with ex. advanced templating I'd really appreciate hearing it!

**Comment by moolen** [comment]
The GCP issue was fixed in f1fad4576cf581d56526450fc8239eb155481f87 (soon to be released, i guess).
